plugins/woocommerce-admin/client/inbox-panel/index.js

```diff
@@ -299,24 +364,33 @@ const InboxPanel = ( { showHeader = true } ) => {
 				/>
 			) }
 			<div className="woocommerce-homepage-notes-wrapper">
-				{ ( isResolvingNotes || isBatchUpdating ) && (
+				{ ! notesHaveResolved && ! allNotes.length && (
 					<Section>
 						<InboxNotePlaceholder className="banner message-is-unread" />
 					</Section>
 				) }
 				<Section>
-					{ ! isResolvingNotes &&
-						! isBatchUpdating &&
+					{ Boolean( allNotes.length ) &&
 						renderNotes( {
+							loadMoreNotes: () => {
+								recordEvent( 'inbox_action_load_more', {
+									quantity_shown: allNotes.length,
+								} );
+								setNoteDisplayQty(
+									noteDisplayQty + ADD_NOTES_AMOUNT
```

Comment:
Instead of relying on the `noteDisplayQty` state to re-trigger the `useSelect` hook, would it make more sense to remove the `useSelect` hook and use `resolveSelect` instead?
This allows you to run the `getNotes` selector just like `apiFetch` and it returns a promise, like so:
`resolveSelect( NOTES_STORE_NAME ).getNotes( inboxQuery ).then( (notes) => setAllNotes(notes)`

I am not opposed to the `useSelect` but given the extra functionality I wonder if `useSelect` would help simplify some of the logic, especially around when to set the all notes state.

Reply:
I've greatly simplified the `useSelect()` in the last commit, so I don't believe we'd see much benefit from this change. I haven't used `resolveSelect()`, so I appreciate the insight!